src/test/ui/async-await/issue-60674.rs

```diff
@@ -0,0 +1,14 @@
+// aux-build:issue-60674.rs
+// compile-pass
+// edition:2018
+#![feature(async_await)]
+
+// This is a regression test that ensures that `mut` patterns are not lost when provided as input
+// to a proc macro.
+
+extern crate issue_60674;
+
+#[issue_60674::attr]
+async fn f(mut x: u8) {}
```

Comment:
@davidtwco Can you follow up with a PR that extends this test to `mut` patterns in nested places?

Reply:
Sure.

Reply:
Submitted #60692.